### PR TITLE
Add kwarg `include_defaults=false` for `to_dict()` and `to_toml()`

### PR DESCRIPTION
**Pull request by chenspc**

### CODE REVIEW

**Comment by chenspc** [comment]
I've given it a try to implement what is discussed in #14 . 

I couldn't find a nice way to combine `args` and `kwargs` for `to_toml` when passing them to `sprint`, so I put all kwargs into a `Dict{Symbo, Any}` as a [workaround](https://github.com/chenspc/Configurations.jl/blob/577e25309dfb0a3b6ec0dbd125201a19c5981be3/src/Configurations.jl#L129). Please let me know if there's a cleaner solution. Thanks.

File: src/Configurations.jl [hunk 1]
```diff
@@ -105,68 +105,77 @@ Curried version of `toml_convert`.
...
 
-Convert an instance `x` of option type to TOML and write it to `String`. See also `TOML.print`.
+`to_toml` does not export fields that are of the same values as the defaults. This can be 
+overridden by changing `include_defaults` to `true`.
 """
-function to_toml(x; sorted=false, by=identity)
-    return sprint(to_toml, x)
+function to_toml(x; sorted=false, by=identity, include_defaults=false)
+    kwargs::Dict{Symbol, Any} = Dict([:sorted => sorted, :by => by, :include_defaults => include_defaults])
+    return sprint(to_toml, x, kwargs)
```
Comment (Roger-luo) [inline src/Configurations.jl:130]: use a closure, and please remove the extra interface you defined for this.

```suggestion
    return sprint(x) do x
        to_toml(x; sorted=sorted, by=>by, include_defaults= include_defaults)
    end
```

**Review by Roger-luo — CHANGES_REQUESTED** [review]
you can just use a closure

**Comment by chenspc** [comment]
I'm not sure how this closure works, but I seem to have got the same error as I had before..

**Pull request closed without merging**